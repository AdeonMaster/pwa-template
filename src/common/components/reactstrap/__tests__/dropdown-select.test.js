import { render } from '@testing-library/react';
import { useState, useCallback } from 'react';

import DropdownSelect from '../dropdown-select';

const options = [
  { value: 'first', displayValue: 'First' },
  { value: 'second', displayValue: 'Second' },
  { value: 'third', displayValue: 'Third' },
];

const WrapperComponent = () => {
  const [value, setValue] = useState(options[1].value);

  const handleChange = useCallback((selectedValue) => setValue(selectedValue), [setValue]);

  return (
    <DropdownSelect placeholder="Select" options={options} value={value} onChange={handleChange} />
  );
};

describe('input-icon-addon', () => {
  it('should render properly', () => {
    const { container } = render(<WrapperComponent />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
