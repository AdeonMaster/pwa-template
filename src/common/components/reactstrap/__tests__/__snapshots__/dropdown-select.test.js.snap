// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input-icon-addon should render properly 1`] = `
<div
  class="dropdown"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="custom-select text-left"
  >
    Second
  </button>
  <div
    aria-hidden="true"
    class="dropdown-menu"
    role="menu"
    tabindex="-1"
  >
    <button
      class="dropdown-item"
      role="menuitem"
      tabindex="0"
      type="button"
    >
      First
    </button>
    <button
      class="dropdown-item active"
      role="menuitem"
      tabindex="0"
      type="button"
    >
      Second
    </button>
    <button
      class="dropdown-item"
      role="menuitem"
      tabindex="0"
      type="button"
    >
      Third
    </button>
  </div>
</div>
`;
